event/event.go

```diff
@@ -0,0 +1,49 @@
+package event
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/bitrise-io/bitrise-step-analytics/models"
+)
+
+type Tracker interface {
+	Send(analytics models.TrackEvent) error
+}
+
+type tracker struct {
+	topic   *pubsub.Topic
+	context *context.Context
+}
+
+func NewTracker(projectID string, topic string) Tracker {
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		panic(fmt.Sprintf("Couldn't start PubSub Client: %s", err.Error()))
+	}
+	return tracker{topic: client.Topic(topic), context: &ctx}
+}
+
+func (t tracker) Send(analytics models.TrackEvent) error {
+	properties := map[string]interface{}{"id": analytics.ID, "ts": convertEpochInMicrosecondsToBigQueryTimestampFormat(analytics.Timestamp), "event_name": analytics.EventName}
+	for k, v := range analytics.Properties {
+		properties[k] = v
```

Comment:
should not we somehow block the client overriding the default properties (`id`, `ts`, ...)?

Reply:
Good point, I'll ignore those properties :)